### ISSUE TITLE
incompatible_disallow_struct_provider_syntax

### ISSUE THREAD

**Issue by drigz** [bazelbuild/rules_go#2012]
rules_go doesn't build with --incompatible_disallow_struct_provider_syntax. https://github.com/bazelbuild/bazel/issues/7347

Example error:

```
ERROR: /rules_go/tests/legacy/build_constraints/BUILD.bazel:10:1: in _cgo_import rule //tests/legacy/build_constraints:go_default_library%linux_amd64%cgo_import:
                                                                                                                                                                                                                   
                                                                                                                                                                                                         
/external/io_bazel_rules_go/go/private/rules/cgo.bzl:363:12: Returning a struct from a rule implementation function is
 deprecated and will be removed soon. It may be temporarily re-enabled by setting --incompatible_disallow_struct_provider_syntax=false . See https://github.com/bazelbuild/bazel/issues/7347 for details.
```

This flag became available with Bazel 0.24, although I haven't seen any information about when it's likely to be flipped. Since Bazel at HEAD is itself incompatible ([source](https://github.com/bazelbuild/bazel/issues/7347#issuecomment-475007269)) I can only guess it won't be flipped in the next release!

**Comment by jayconrod** [comment]
I'll take a look at this before the next set of releases. I fixed some of the incompatibility warnings in #1978, but the fact that `@bazel_tools` breaks with these flags on makes it difficult to find code other code that will be broken.
